pyensembl/database.py

```diff
@@ -83,33 +83,42 @@ def _create_database(self):
             table_name="ensembl",
             df=df,
             subdir=CACHE_SUBDIR,
-            overwrite=False,
+            overwrite=force,
             indices=indices)
         return db
 
-    def _connect_or_create_database(self):
+    def _connect_if_exists(self):
         """
-        If database already exists, open a connection.
-        Otherwise, create it.
+        Return the connection if the DB exists, and otherwise return 
+        None. As a side effect, stores the database connection in 
+        self._connection.
         """
-        db_path = self.local_db_path()
-        if exists(db_path):
-            db = sqlite3.connect(db_path)
-            # maybe file got created but not filled
-            if datacache.db.db_table_exists(db, 'ensembl'):
-                return db
-        return self._create_database()
+        if self._connection is None:
+            db_path = self.local_db_path()
+            if exists(db_path):
+                connection = sqlite3.connect(db_path)
+                # maybe file got created but not filled
+                if datacache.db.db_table_exists(connection, 'ensembl'):
+                    self._connection = connection
+        return self._connection
 
     @property
     def connection(self):
         """
         Return the sqlite3 database for this Ensembl release
-        (download and/or construct it necessary).
-        As a side effect, stores the database connection in self._db
+        (download and/or construct it if necessary, if auto_download
+        is on). As a side effect, stores the database connection in 
+        self._connection.
         """
-        if self._connection is None:
-            self._connection = self._connect_or_create_database()
-        return self._connection
+        if self._connect_if_exists():
+            return self._connection
+        if self.gtf.auto_download:
```

Comment:
>I think these code paths are too fragile and confusing.

I agree, I was already uncomfortable with the mental complexity of these paths and now it seems even trickier to follow. However, I'd suggest merging this PR and then thinking about paring back the complexity when we get frustrated with it later.